src/sparkle/reader/kafka_reader.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any
+from pyspark.sql import SparkSession, DataFrame
+from sparkle.config import Config
+from sparkle.reader.schema_registry import SchemaRegistry
+from sparkle.utils.spark import parse_by_avro
+
+
+class KafkaReader:
+    """KafkaReader is a reader for streaming data from Kafka using Spark.
+
+    This class allows you to read data from a specified Kafka topic, with support
+    for Avro format parsing using a schema registry.
+
+    Attributes:
+        topic (str): Kafka topic to read from.
+        schema_registry (SchemaRegistry): Schema registry client for fetching Avro schemas.
+        schema_version (str): Version of the schema to use for Avro parsing.
+        use_avro (bool): Flag indicating whether to parse data using Avro format.
+        kafka_options (dict): Dictionary containing Kafka configuration options for Spark.
+    """
+
+    def __init__(
+        self,
+        spark: SparkSession,
+        topic: str,
+        schema_registry: SchemaRegistry,
+        use_avro: bool = True,
```

Comment:
Can we move this to `schema_retgistry` config object? And the key can be `format_`, and if it's of type ‍`avro` (string or Enum), the reader will act accordingly.